src/User/src/Repository/UserRepository.php

```diff
@@ -239,4 +245,47 @@ public function saveUser(User $user): void
         $this->getEntityManager()->persist($user);
         $this->getEntityManager()->flush();
     }
+
+    public function getUserEntityByUserCredentials(
+        $username,
+        $password,
+        $grantType,
+        ClientEntityInterface $clientEntity
+    ): ?UserEntity {
+        $qb = $this->getEntityManager()->createQueryBuilder();
+        switch ($clientEntity->getName()) {
+            case 'admin':
```

Comment:
You could use here, on line **263** and **285** the `OauthClient::NAME_*` constants.